### ISSUE TITLE
[NV-2290] 🐛 Bug Report: Email template `{{#if var}}` incorrectly resolves to boolean with default `true`

### ISSUE THREAD

**Issue by pxue** [novuhq/novu#3387]
### 📜 Description

There's actually two bugs here

1. When an email contains conditional check such as `{{#if var}}`, the variable is incorrectly parsed as `boolean` with the default value set to `true`. `var` can be any "falsy" value such as empty strings or empty nested json objects.

2. After changing the default value to `false`, the value can be reset if you make other modifications. As far as i can tell, promoting to production also resets the default value back to True

### 👟 Reproduction steps

1. Create an email notificaiton
2. use custom code:

```handlebars
{{#if var}}
we sent {{var}}
{{/if}}
```

4. ensure `var` is boolean with default value `true`
![Screenshot 2023-05-10 at 5 43 01 PM](https://github.com/novuhq/novu/assets/270494/271d640a-c6d2-441b-af5a-66a85a567b9d)

5. Even after setting the default value to `false`, after saving the template with changes, it reverts back to `true`

6. Send an empty payload:

```json
{
	"name": "test-notification",
	"to": { "subscriber": "subscriber:123" },
	"payload": {},
	"transactionId": null,
	"actor": "subscriber:165"
}
```

### 👍 Expected behavior

- Boolean value should not have the default value `true`

### 👎 Actual Behavior with Screenshots

![Screenshot 2023-05-10 at 5 44 40 PM](https://github.com/novuhq/novu/assets/270494/2c50cbb6-6b55-46c4-9cfd-3b3f45222216)

and email,

![Screenshot 2023-05-10 at 5 46 02 PM](https://github.com/novuhq/novu/assets/270494/9d1c053e-5f03-4b48-8614-5d9f8fea7925)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

<sub>[NV-2290](https://linear.app/novu/issue/NV-2290/🐛-bug-report-email-template-if-var-incorrectly-resolves-to-boolean)</sub>

**Comment by gitstart** [comment]
@scopsy I would like to pick this up

**Comment by ainouzgali** [comment]
Hi @pxue, if you can, please check this one again with the latest version: 

> 2\. After changing the default value to `false`, the value can be reset if you make other modifications. As far as i can tell, promoting to production also resets the default value back to True

I believe the promotion part has been fixed in v0.16.0. About the reset - it is a different bug, reproduced by clicking update and then reopening the variable modal "resets" the variable values. I will open a separate issue for that, as it is related to all variables, not only booleans. 



> * Boolean value should not have the default value `true`

IMO, that really depends. I feel it is right to go with true as the default value. As we don't know the context of why the user creates the variable it is fair to assume to default it to true unless changed by the user. 


Until we fix the bug I mentioned, if you change your default value to false as your last action and then click update, that should work for you and not reset. And as I said, the promotion was fixed in v0.16.0

**Comment by scopsy** [comment]
Hi @pxue did you had a chance to see if that was resolved for you as Gali suggested in the new version? Want to check if we can close this issue or should investigate this more from our side 🙏

**Comment by pxue** [comment]
@scopsy @ainouzgali 

Apologies for the delay.
Confirming that the promotion part is indeed fixed. Happy to continue the other bugs in another ticket.

Thanks!

**Comment by ainouzgali** [comment]
Just updating that the reset bug has been fixed in v0.17.0 @pxue :)
